### PR TITLE
Tulip V1

### PR DESCRIPTION
**Pull request by eddya101**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [YES ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ YES] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ YES] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [YES ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [YES ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [YES ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [N/A ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [YES ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ YES] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ Not required] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md. 

Power Automate Flows:
Screenshot 2023-01-19 at 16 47 26
Screenshot 2023-01-19 at 17 02 30
Screenshot 2023-01-19 at 16 57 30
Screenshot 2023-01-19 at 17 02 15

Flow Builder Testing:
make powerautomate com_environments_Default-95c64e18-c009-4e46-bdd9-ecfc373597e4_connections_available_custom_shared_tulip-20connector-20test-2016-20jan-5f3e1fcf7ed-e1d30bedbc48887d_edit_test

### CODE REVIEW

**Comment by eddya101** [comment]
Re API Properties Validator Fail - The "scriptOperations" property was added by customer connectors editing tool when I added the code. Anty advice how to fix this one?

File: certified-connectors/Tulip/readme.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+The Tulip Connector allows you to integrate with the data in your Tulip Instance. Enxtending your Tulip frontline operations applications into your power automate, Teams and M365 workflows.
+
+## Prerequisites
+You will need the following to proceed:
+* Active Tulip instance and subscription (tulip.co)
+* Tulip API bot credentials for your Tulip Account with access to the appropriate scopes. ( Tulip Account or Workspace owner can provide credentials)
+
+## Setting up the connector
+On initial use, you will need to enter your Tulip Bot Credentials and Tulip instance url. This will provide you with the connection between power automate and you Tulip instance. See [Adding an API](https://support.tulip.co/docs/how-to-use-the-table-api-1#:~:text=in%20a%20table.-,Adding,-an%20API) to see how to create a bot in your Tulip instance.
+Then  you will beable to configure the connector with in your power automate flow.
```
Comment (Amjed-Ayoub) [inline certified-connectors/Tulip/readme.md:12]: @eddya101 all files should be free of grammatical and spelling errors. Please fix "beable" to "be able".

File: certified-connectors/Tulip/readme.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+note: if / where using the create record you will need to create with a uique id
```
Comment (Amjed-Ayoub) [inline certified-connectors/Tulip/readme.md:13]: @eddya101 all files should be free of grammatical and spelling errors. Please fix "uique" to "unique".

File: certified-connectors/Tulip/readme.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+
+## Supported Operations
+The connector supports the following operations:
+* `Get Record`:  Returns a record from a Tulip Table based on a provided record id
+* `Create Record`: Creates a new record in a Tulip Table. Note: requires a unique id to be defined.
+* `Put Record`: Updates a record in a Tulip Table
+
+## Known Issues & Limitation
+* Files Type - The connector is currently limited to string, numbers, integers and boolens. It will not handle images, videos and files.
```
Comment (Amjed-Ayoub) [inline certified-connectors/Tulip/readme.md:23]: @eddya101 all files should be free of grammatical and spelling errors. Please fix "boolens" to "booleans".

File: certified-connectors/Tulip/readme.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+### Tulip Connector
+
+The Tulip Connector allows you to integrate with the data in your Tulip Instance. Enxtending your Tulip frontline operations applications into your power automate, Teams and M365 workflows.
```
Comment (Amjed-Ayoub) [inline certified-connectors/Tulip/readme.md:3]: @eddya101 all files should be free of grammatical and spelling errors. Please fix "Enxtending" to "Extending".

File: certified-connectors/Tulip/settings.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (Amjed-Ayoub) [inline certified-connectors/Tulip/settings.json:1]: @eddya101 Settings.json is not required in open source repository. Please remove this file from the pull request. You can submit this file later in ISV Studio.

File: custom-connectors/Tulip/readme.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+### Tulip Connector
```
Comment (Amjed-Ayoub) [inline custom-connectors/Tulip/readme.md:1]: @eddya101 Please remove all the custom connector files from this PR. You can submit them with a separate pull request. We will approve and merge it. Thanks.

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]
@eddya101 thank you so much for your submission. The icon is not allowed in the open source repository. Please remove all your connector icons from this PR.
You can include the below files.

1. apiDefinition.swagger.json
2. apiProperties.json
3. readme.md
4. script.csx(optional in open source repository)

Thank you so much.

**Comment by Amjed-Ayoub** [comment]
> Re API Properties Validator Fail - The "scriptOperations" property was added by customer connectors editing tool when I added the code. Anty advice how to fix this one?

@eddya101 Please ignore this error.

**Comment by eddya101** [comment]
@microsoft-github-policy-service agree [company="Tulip Interfaces"]

**Comment by eddya101** [comment]
@microsoft-github-policy-service agree company="Tulip Interfaces"

**Comment by eddya101** [comment]
@Amjed-Ayoub Thanks for your review and feedback. All comments are now addressed. Let me know if there is anything else.

File: custom-connectors/Tulip/readme.md (hunk 6 above)
Comment (eddya101) [reply custom-connectors/Tulip/readme.md]: @Amjed-Ayoub All files are now committed. Do you want me to close this pull request and create a new one? Or are you able to approve this one?

**Review by Amjed-Ayoub — APPROVED** [review]

**Comment by Amjed-Ayoub** [comment]
Hello @eddya101,
I hope you are doing well.
Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) or [update your submission](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-updates) using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file.
After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6.
Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector. 
We expect all tests to be completed within 1-2 weeks.
If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions.
Please let me know if you have any questions or run into any problems while creating or updating your submission.
Thank you very much for working with us.
